build/utils/npm_test.go

```diff
@@ -192,12 +195,92 @@ func TestDependencyWithNoIntegrity(t *testing.T) {
+	npmVersion, _, err := GetNpmVersionAndExecPath(logger)
+	require.NoError(t, err)
+	if !npmVersion.AtLeast("7.0.0") {
+		t.Skip("Running on npm v7 and above only, skipping...")
+	}
+	path, cleanup := testdatautils.CreateTestProject(t, filepath.Join("..", "testdata/npm/project6"))
+	defer cleanup()
+	data, err := os.ReadFile(filepath.Join(path, "package-lock_test.json"))
+	require.NoError(t, err)
+	info, err := os.Stat(filepath.Join(path, "package-lock_test.json"))
+	require.NoError(t, err)
+	require.NoError(t, os.WriteFile(filepath.Join(path, "package-lock.json"), data, info.Mode().Perm()))
+	// sleep so the package.json modified time will be bigger than the package-lock.json, this make sure it will recalculate lock file.
+	time.Sleep(time.Millisecond * 20)
+	require.NoError(t, os.Chtimes(filepath.Join(path, "package.json"), time.Now(), time.Now()))
+
+	// Calculate dependencies.
+	dependencies, err := CalculateDependenciesMap("npm", path, "jfrogtest",
+		NpmTreeDepListParam{Args: []string{}, IgnoreNodeModules: true, OverWritePackageLock: true}, logger)
+	assert.NoError(t, err)
+	expectedRes := map[string]*dependencyInfo{
+		"underscore:1.13.6": {
+			Dependency: entities.Dependency{
+				Id:          "underscore:1.13.6",
+				Scopes:      []string{"prod"},
+				RequestedBy: [][]string{{"jfrogtest"}},
+				Checksum:    entities.Checksum{},
+			},
+			npmLsDependency: &npmLsDependency{
+				Name:      "underscore",
+				Version:   "1.13.6",
+				Integrity: "sha512-+A5Sja4HP1M08MaXya7p5LvjuM7K6q/2EaC0+iovj/wOcMsTzMvDFbasi/oSapiwOlt252IqsKqPjCl7huKS0A==",
+			},
+		},
+		"cors.js:0.0.1-security": {
+			Dependency: entities.Dependency{
+				Id:          "cors.js:0.0.1-security",
+				Scopes:      []string{"prod"},
+				RequestedBy: [][]string{{"jfrogtest"}},
+				Checksum:    entities.Checksum{},
+			},
+			npmLsDependency: &npmLsDependency{
+				Name:      "cors.js",
+				Version:   "0.0.1-security",
+				Integrity: "sha512-Cu4D8imt82jd/AuMBwTpjrXiULhaMdig2MD2NBhRKbbcuCTWeyN2070SCEDaJuI/4kA1J9Nnvj6/cBe/zfnrrw==",
+			},
+		},
+		"lightweight:0.1.0": {
+			Dependency: entities.Dependency{
+				Id:          "lightweight:0.1.0",
+				Scopes:      []string{"prod"},
+				RequestedBy: [][]string{{"jfrogtest"}},
+				Checksum:    entities.Checksum{},
+			},
+			npmLsDependency: &npmLsDependency{
+				Name:      "lightweight",
+				Version:   "0.1.0",
+				Integrity: "sha512-10pYSQA9EJqZZnXDR0urhg8Z0Y1XnRfi41ZFj3ZFTKJ5PjRq82HzT7LKlPyxewy3w2WA2POfi3jQQn7Y53oPcQ==",
+			},
+		},
+		"minimist:0.1.0": {
+			Dependency: entities.Dependency{
+				Id:          "minimist:0.1.0",
+				Scopes:      []string{"prod"},
+				RequestedBy: [][]string{{"jfrogtest"}},
+				Checksum:    entities.Checksum{},
+			},
+			npmLsDependency: &npmLsDependency{
+				Name:      "minimist",
+				Version:   "0.1.0",
+				Integrity: "sha512-wR5Ipl99t0mTGwLjQJnBjrP/O7zBbLZqvA3aw32DmLx+nXHfWctUjzDjnDx09pX1Po86WFQazF9xUzfMea3Cnw==",
+			},
+		},
+	}
```

Comment:
For better test readability, is it possible to export this to a const outside of the test method?